File: foxspot/src/main/kotlin/com/example/foxspot/model/Event.kt
//package com.example.foxspot.model
//
//import org.springframework.data.annotation.Id
//import org.springframework.data.mongodb.core.mapping.Document
//
//@Document(collection = "events")
//class Event(
//    val name: String,
//    val location: Point
//): Base()
package com.example.foxspot.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "events")
class Event(
        val name: String,
        val address: String,
        val city: String,
        val eventType: String,
        val description: String,
        val image: String,
        val location: Point
) : Base()

File: mobile-app/model/event.ts
import { Point } from "@/model/point";
import {BaseModel} from "@/model/base";

export interface Event extends BaseModel {
    name?: string;
    location?: Point;
    address?: string;
    city?: string;
    eventType?: string;
    description?: string;
    image?: string;
    tags?: string[];
    startTime?: string;
    endTime?: string;
    status?: string;
}
